### PR TITLE
Improve password masking utility

### LINKED ISSUE #605: PostgreSQL passwords are logged in plain text (as part of DbUri)
Here’s an example (where the password does get obfuscated in the unused defaults (`User:postgres password: ****`), but not in the `DbUri`):

```
promscale[578194]: level=info ts=2021-04-28T23:30:36.119Z caller=runner.go:33 config="&{ListenAddr:[::1]:9201 PgmodelCfg:{CacheConfig:{SeriesCacheInitialSize:250000 seriesCacheMemoryMaxFlag:{kind:0 value:50} SeriesCacheMemoryMaxBytes:54001038131 MetricsCacheSize:10000 LabelsCacheSize:10000} AppName:promscale@0.3.0 Host:localhost Port:5432 User:postgres password: ****Database:timescale SslMode:require DbConnectRetries:0 DbConnectionTimeout:1m0s AsyncAcks:false ReportInterval:0 WriteConnectionsPerProc:4 MaxConnections:-1 UsesHA:false DbUri:postgres:///db_promscale?user=pguser_promscale&password=<<OUCH_PLAINTEXT_PASSWORD>>&host=/run/postgresql/&port=5432&sslcert=/etc/promscale/postgres_client.crt&sslkey=/etc/promscale/postgres_client.key&sslmode=verify-full EnableStatementsCache:****"
```

It would be nice to always wipe out passwords from log output.

For context, I’ve noticed the problem while debugging [this issue](https://bugs.archlinux.org/task/70633). However, password logging ocurs no matter if `promscale` starts successfully or not.

### PR DESCRIPTION
**Pull request by antekresic**
Previous implementation did not cover all possible cases that needed.
This change also reduces the number of regexes necessary to mask all
combinations.

Fixes #605

### CODE REVIEW

**Review by cevian — APPROVED** [review]

File: pkg/util/util_test.go [hunk 1]
```diff
@@ -65,34 +65,40 @@ func TestThroughputCalc(t *testing.T) {
...
-		"host: localhost password: foobar":                                   "host: localhost password: ****",
-		"host: localhost Password: foobar":                                   "host: localhost password: ****",
-		"postgres://postgres:password@localhost:5432/postgres?sslmode=allow": "postgres://postgres:****@localhost:5432/postgres?sslmode=allow",
+		"password='foobar' host='localhost'":                                                   "password='****' host='localhost'",
+		"password='foobar&' host='localhost'":                                                  "password='****' host='localhost'",
+		"password='foo bar' host='localhost'":                                                  "password='****' host='localhost'",
+		"Password='foo bar' host='localhost'":                                                  "password='****' host='localhost'",
+		"password='foo'bar' host='localhost'":                                                  "password='****'bar' host='localhost'",
+		"password= 'foobar' host='localhost'":                                                  "password= '****' host='localhost'",
```
Comment (Harkishen-Singh) [inline pkg/util/util_test.go:73]: Is space before '* necessary in masking? Or is it the restrain due to implement method?

File: pkg/util/util_test.go [hunk 2]
```diff
@@ -65,34 +65,40 @@ func TestThroughputCalc(t *testing.T) {
...
+		"password=  'foobar' host='localhost'":                                                 "password=  '****' host='localhost'",
+		"pass='foobar' host='localhost'":                                                       "pass='foobar' host='localhost'",
+		"host='localhost' password='foobar'":                                                   "host='localhost' password='****'",
+		"password:foobar  host: localhost":                                                     "password:****  host: localhost",
+		"password:foo bar  host: localhost":                                                    "password:****  host: localhost",
+		"password: foobar  host: localhost":                                                    "password: ****  host: localhost",
+		"password:  foobar  host: localhost":                                                   "password:  ****  host: localhost",
```
Comment (Harkishen-Singh) [inline pkg/util/util_test.go:80]: Similarly, trailing space after masking?

File: pkg/util/util_test.go (hunk 1 above)
Comment (antekresic) [reply pkg/util/util_test.go:73]: Respects previous spacing.

File: pkg/util/util_test.go (hunk 2 above)
Comment (antekresic) [reply pkg/util/util_test.go:80]: Respects previous spacing.

**Pull request merged**